catalog/plugin/catalog-plugin-auditclassifiedaccess/src/main/java/org/codice/alliance/catalog/plugin/auditclassified/AuditClassifiedAccessPlugin.java

```diff
@@ -0,0 +1,179 @@
+    LOGGER.info("Response went through the Audit Classified Access Plugin");
+    return input;
+  }
+
+  private void handleAudits(Metacard metacard) {
+    auditClassification(metacard);
+    auditReleasability(metacard);
+    auditDisseminationControls(metacard);
+    auditCodewords(metacard);
+  }
+
+  private void auditClassification(Metacard metacard) {
+    Attribute classificationAttribute = metacard.getAttribute(Security.CLASSIFICATION);
+    if (classificationAttribute != null) {
+      // NOTE: Single-valued attribute so being handled differently at the moment
+      String classificationValue = (String) classificationAttribute.getValue();
+      if (classificationValue != null) {
+        for (String enteredClassificationValue : getEnteredClassificationValues()) {
+          if (classificationValue.equals(enteredClassificationValue)) {
+            SecurityLogger.audit(
+                "Audit "
+                    + Security.CLASSIFICATION
+                    + " attribute with value "
+                    + enteredClassificationValue
+                    + " on metacard "
+                    + metacard.getId());
+          }
+        }
+      }
+    }
+  }
+
+  private void auditReleasability(Metacard metacard) {
+    List<Serializable> releasabilityValues = getAttributeValues(metacard, Security.RELEASABILITY);
+    if (releasabilityValues != null) {
+      auditValues(releasabilityValues, Security.RELEASABILITY, metacard.getId());
+    }
+  }
+
+  private void auditDisseminationControls(Metacard metacard) {
+    List<Serializable> disseminationControlsValues =
+        getAttributeValues(metacard, Security.DISSEMINATION_CONTROLS);
+    if (disseminationControlsValues != null) {
+      auditValues(disseminationControlsValues, Security.DISSEMINATION_CONTROLS, metacard.getId());
+    }
+  }
+
+  private void auditCodewords(Metacard metacard) {
+    List<Serializable> codewordsValues = getAttributeValues(metacard, Security.CODEWORDS);
+    if (codewordsValues != null) {
+      auditValues(codewordsValues, Security.CODEWORDS, metacard.getId());
+    }
+  }
+
+  private List<Serializable> getAttributeValues(Metacard metacard, String attributeString) {
+    List<Serializable> attributeValues = null;
+    Attribute attribute = metacard.getAttribute(attributeString);
+    if (attribute != null) {
+      attributeValues = attribute.getValues();
+    }
+    return attributeValues;
+  }
+
+  private void auditValues(
+      List<Serializable> valuesToCheckToAudit, String attributeString, String metacardId) {
+
+    List<String> enteredValues = null;
+
+    if (attributeString.equals(Security.RELEASABILITY)) {
+      enteredValues = getEnteredReleasabilityValues();
+    }
+
+    if (attributeString.equals(Security.DISSEMINATION_CONTROLS)) {
+      enteredValues = getEnteredDisseminationControlsValues();
+    }
+
+    if (attributeString.equals(Security.CODEWORDS)) {
+      enteredValues = getEnteredCodewordsValues();
+    }

```

Comment:
✏️ **[Map]**: You wouldn't need each of these if statements for the generic method because it'll just get the values from your map based on the `attributeString` passed in.